### PR TITLE
Add query env

### PR DESCRIPTION
**Pull request by rosamakinen**
Added QUERY_STRING to cgi env variables and the surrounding functionality to it. Small tinkering with the hello_world.py, to try to make it work. 

Next step would be playing more with the script/cgi responsse to make it actually output the correct thing.

### CODE REVIEW

**Review by forbidden-arts — APPROVED** [review]
minor merge conflict, but otherwise looks okay

**Comment by rosamakinen** [comment]
> minor merge conflict, but otherwise looks okay

Oh, right. I had made this branch from the main yesterday, since the PR that was hanging from you and meri had some issues with the method types and would give bad method  -error at all times and I was not sure when that would be solved and it merged. I will obv pull the new main and solve the merge conflicts before this gets merged to the main, so no worries.

File: src/CgiHandler.cpp [hunk 1]
```diff
@@ -1,51 +1,70 @@
...
+	cgiEnvironment["REQUEST_METHOD"] = request.translateMethod(request.getMethod());
+	cgiEnvironment["SCRIPT_FILENAME"] = FileHandler::getFilePath(request.getUri());
+	cgiEnvironment["SERVER_SOFTWARE"] = "SillyLittleSoftware/1.0"; //fetch from config?
+	cgiEnvironment["SERVER_NAME"] = "127.0.0.1"; //fetch from config?
+	cgiEnvironment["QUERY_STRING"] = transferToString(request.getParameters());
+
+	return cgiEnvironment;
+}
+
+std::string findInterpreterPath(std::string fullPath)
```
Comment (merituulie) [inline src/CgiHandler.cpp:57]: does this mean we could actually run other scripts than python too?

Comment (merituulie) [reply src/CgiHandler.cpp:57]: I mean that this finds basically which command we are running aka given bash it would get that too? nice!

**Review by merituulie — APPROVED** [review]

File: src/CgiHandler.cpp (hunk 1 above)
Comment (rosamakinen) [reply src/CgiHandler.cpp:57]: in theory yes, if we add the other validation (and whatever config would need?) for the correct extension. Then we could just add the working scripts and make sure they do the response correctly.

**Pull request merged**